dora/core/server/worker/src/main/java/alluxio/worker/modules/WorkerIdentityProvider.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.modules;
+
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.PropertyKey;
+import alluxio.wire.WorkerIdentity;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * Provider that resolves worker identities from configuration, persisted storage, or
+ * automatically generates a new one.
+ */
+public class WorkerIdentityProvider implements Provider<WorkerIdentity> {
+  private static final Logger LOG = LoggerFactory.getLogger(WorkerIdentityProvider.class);
+  private static final String WORKER_IDENTITY_FILE = "worker_identity";
+  private final AlluxioConfiguration mConf;
+
+  /**
+   * @param conf configuration
+   */
+  @Inject
+  public WorkerIdentityProvider(AlluxioConfiguration conf) {
+    mConf = conf;
+  }
+
+  /**
+   * Resolves a worker's identity from the following sources, in the order of preference:
+   * <ol>
+   *   <li>
+   *     Alluxio configuration. This includes JVM system properties, alluxio-site.properties, etc.
+   *   </li>
+   *   <li>
+   *     Persisted storage. A identity file is persisted in the worker process's working directory.
+   *   </li>
+   *   <li>
+   *     Automatically generated as a UUID.
+   *   </li>
+   * </ol>
+   *
+   * @return worker identity
+   */
+  @Override
+  public WorkerIdentity get() {
+    // Look at configurations first
+    if (mConf.isSetByUser(PropertyKey.WORKER_IDENTITY_UUID)) {
+      String uuidStr = mConf.getString(PropertyKey.WORKER_IDENTITY_UUID);
+      return WorkerIdentity.ParserV1.INSTANCE.fromUUID(uuidStr);
```

Comment:
if we allow setting workid from site properties does that mean all workers' site.properties are not the same? that might cause bit of maintenance overhead?

Reply:
Does not have to be set in `alluxio-site.properties`. The best way is through a environment variable. I imagine in Kubernetes users want to centrally manage the worker identities, so they can utilize a k8s runtime configuration to set a different identity for each worker instance.